tests/rptest/tests/pandaproxy_test.py

```diff
@@ -834,12 +834,28 @@ def test_consumer_group_binary_v2(self):
 
         # Fetch from a consumer
         self.logger.info(f"Consumer fetch")
-        cf_res = c0.fetch()
-        assert cf_res.status_code == requests.codes.ok
-        fetch_result = cf_res.json()
+        fetch_result = []
+
+        def do_fetch():
+            cf_res = c0.fetch()
+            assert cf_res.status_code == requests.codes.ok
+            records = cf_res.json()
+            self.logger.info(f"Fetched {len(records)} records: {records}")
+            fetch_result.extend(records)
+            self.logger.info(
+                f"Have {len(fetch_result)} records?: {fetch_result}")
+            if len(fetch_result) != 9:
+                self.logger.info(f"Fetch Mitigation {len(fetch_result)}")
+            return len(fetch_result) == (3 * 3)
+
+        wait_until(lambda: do_fetch(),
+                   timeout_sec=10,
+                   backoff_sec=0,
+                   err_msg="Timeout waiting for records to appear")
+
+        self.logger.info(f"Fetched: {fetch_result}")
         # 3 topics * 3 msg
-        assert len(fetch_result) == 3 * 3
-        print(fetch_result)
+        assert len(fetch_result) == (3 * 3)
```

Comment:
What is the purpose of this assert? Doesn't it accomplish the same thing as line 849?

Reply:
I think I wasn't sure that a failing `wait_until` fails the test.

Reply:
If wait_until fails, it throws a TImeoutError. It looks like your's will raise with `TimeoutError("Timeout waiting for records to appear")`